scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -425,7 +425,9 @@
                                                (if-let [compute-cluster (cc/compute-cluster-name->ComputeCluster compute-cluster-name)]
                                                  (do (log/info "Attempting to kill task" task-id "in" compute-cluster-name "due to job completion")
                                                      (meters/mark! tx-report-queue-tasks-killed)
-                                                     (cc/safe-kill-task compute-cluster task-id))
+                                                     (cc/safe-kill-task compute-cluster task-id)
```

Comment:
Sorry I didn't catch this earlier, but shouldn't we do the db update first, then kill. Then we're consistent with how it works in cancelled tasks?

Reply:
OK. but why do we do it in that order for the cancelled-task-killer
maybe we should flip  the order there

Reply:
We'd be churning the reason twice with the race. We'd see the executor die, put that reason in, then we'd be replacing it with killed-by-user.